credsweeper/app.py

```diff
@@ -250,19 +251,30 @@ def file_scan(self, content_provider: ContentProvider) -> List[Candidate]:
                                                             content_provider.file_type, content_provider.info)
             candidates.append(dummy_candidate)
 
-        elif self.config.depth > 0 and isinstance(content_provider, TextContentProvider):
-            # Feature to scan files which might be containers
-            data = Util.read_data(content_provider.file_path)
-            if data:
-                data_provider = DataContentProvider(data=data,
-                                                    file_path=content_provider.file_path,
-                                                    info=content_provider.file_path)
-                candidates = self.data_scan(data_provider, self.config.depth, RECURSIVE_SCAN_LIMITATION)
-
         else:
             # Regular file scanning
-            analysis_targets = content_provider.get_analysis_target()
-            candidates = self.scanner.scan(analysis_targets)
+            if content_provider.file_type not in self.config.exclude_containers:
+                analysis_targets = content_provider.get_analysis_target()
+                candidates.extend(self.scanner.scan(analysis_targets))
+
+            if self.config.depth and isinstance(content_provider, TextContentProvider):
+                # Feature to scan files which might be containers
+                data = Util.read_data(content_provider.file_path)
+                if data:
+                    data_provider = DataContentProvider(data=data,
+                                                        file_path=content_provider.file_path,
+                                                        file_type=content_provider.file_type,
+                                                        info=content_provider.file_path)
+                    extra_candidates = self.data_scan(data_provider, self.config.depth, RECURSIVE_SCAN_LIMITATION)
+                    if extra_candidates:
+                        # reduce duplicated credentials
+                        found_values = set(line_data.value for candidate in candidates
+                                           for line_data in candidate.line_data_list)
+                        for i in extra_candidates:
+                            for j in i.line_data_list:
+                                if j.value not in found_values:
+                                    candidates.append(i)
```

Comment:
Using `i` and `j` is not good... we already using full name of variable..

```suggestion
                        for extra_candidate in extra_candidates:
                            for line_data in i.line_data_list:
                                if line_data.value not in found_values:
                                    candidates.append(extra_candidate)
```